scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -146,6 +150,12 @@
 	/** The next expected handshake message sequence number. */
 	private int nextReceiveSeq = 0;
 
+	/** The current flight number. */
+	protected int flightNumber = 0;
+
+	private int maxDeferredProcessedApplicationDataMessages;
+	private List<Object> deferredApplicationData;
```

Comment:
You could make `maxDeferredProcessedApplicationDataMessages` && `deferredApplicationData` final,

Or implements `takeDeferredApplicationData`
like this :
```java
List<Object> applicationData = deferredApplicationData;
deferredApplicationData = new new ArrayList<Object>();
return applicationData;
```

Reply:
I just don't like that kind of "takes" :-). 
The finals are added.